src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/Eventing/AsyncParameterCapturingEventSource.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.StartupHook;
+using System;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Microsoft.Diagnostics.Monitoring.HostingStartup.ParameterCapturing.Eventing
+{
+    internal sealed class AsyncParameterCapturingEventSource : IDisposable
+    {
+        private readonly Thread _thread;
+        private readonly CancellationTokenSource _cts = new();
+        private readonly BlockingCollection<Action<CancellationToken>> _pendingEvents;
+        private readonly ParameterCapturingEventSource _eventSource;
+
+        private const int MaxPendingEventsCollectionCapacity = 1024;
+        private const string BackgroundLoggingThreadName = "[dotnet-monitor] Probe EventSource Thread";
+
+        public AsyncParameterCapturingEventSource(ParameterCapturingEventSource eventSource)
+        {
+            _pendingEvents = new BlockingCollection<Action<CancellationToken>>(MaxPendingEventsCollectionCapacity);
+            _eventSource = eventSource;
+
+            _thread = new(ThreadLoop)
+            {
+                Priority = ThreadPriority.BelowNormal,
+                IsBackground = true,
+                Name = BackgroundLoggingThreadName
+            };
+            _thread.Start();
+        }
+
+        public void Dispose()
```

Comment:
This will throw if called multiple times, lets use our existing `DisposableHelper` class here to handle this.

Example:
https://github.com/dotnet/dotnet-monitor/blob/81667980f8fc0212d1cb0c1727e5b4922accb185/src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/ParameterCapturingLogger.cs#L158-L166